src/main/java/io/strimzi/kafka/metrics/PrometheusMetricsReporterConfig.java

```diff
@@ -60,18 +58,64 @@ private Pattern compileAllowlist(List<String> allowlist) {
         return Pattern.compile(joined);
     }
 
+    public String listener() {
+        return listener.toString();
+    }
+
+    static class Listener {
+
+        private static final Pattern PATTERN = Pattern.compile("http://\\[?([0-9a-zA-Z\\-%._:]*)]?:([0-9]+)");
+
+        final String host;
+        final int port;
+
+        public Listener(String host, int port) {
+            this.host = host;
+            this.port = port;
+        }
+
+        public static Listener parseListener(String listener) {
+            Matcher matcher = PATTERN.matcher(listener);
+            if (matcher.find()) {
+                String host = matcher.group(1);
+                int port = Integer.parseInt(matcher.group(2));
+                return new Listener(host, port);
+            } else {
+                throw new ConfigException("Invalid listener format: " + listener);
```

Comment:
Can we use the constructor with 3 arguments?
https://kafka.apache.org/36/javadoc/org/apache/kafka/common/config/ConfigException.html#%3Cinit%3E(java.lang.String,java.lang.Object,java.lang.String)

Reply:
Do you mean like this?
```throw new ConfigException("Invalid listener format: " + value + ". Must be of format http:// [host]:[port]");```